options/posix/generic/lookup.cpp

```diff
@@ -0,0 +1,135 @@
+#include <mlibc/lookup.hpp>
+#include <mlibc/debug.hpp>
+#include <bits/ensure.h>
+
+#include <frg/string.hpp>
+#include <mlibc/allocator.hpp>
+#include <string.h>
+#include <errno.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+
+namespace mlibc {
+
+static frg::string<MemoryAllocator> read_dns_name(char *buf, char *&it) {
+	frg::string<MemoryAllocator> res{getAllocator()};
+	while (true) {
+		char code = *it++;
+		if ((code & 0xC0) == 0xC0) {
+			// pointer
+			uint8_t offset = ((code & 0x3F) << 8) | *it++;
+			auto offset_it = buf + offset;
+			return res + read_dns_name(buf, offset_it);
+		} else if (!(code & 0xC0)) {
+			if (!code)
+				break;
+
+			for (int i = 0; i < code; i++)
+				res += (*it++);
+
+			if (*(it + 1))
+				res += '.';
+		} else {
+			break;
+		}
+	}
+
+	return res;
+}
+
+int __lookup_name(struct address *buf, const char *name) {
+	mlibc::infoLogger() << "__lookup_name() called!" << frg::endlog;
+	frg::string<MemoryAllocator> request{getAllocator()};
```

Comment:
Using a vector instead of a string might be more appropriate.

Reply:
If the vector misses some required methods (and it's a lot of work to add them), it's OK to go with the string for now.

Reply:
Well I use an `operator+=(const char *)` and im not sure how to implement that for the general case in `frg::vector` (maybe some sort of `frg::span` that we can use for this in general?). So for now I think leaving it as a string is better but I agree that a vector is a better fit.